### PR TITLE
Exposed context resize and setTransform

### PR DESCRIPTION
**Pull request by sjrct**

### CODE REVIEW

File: tealeaf/js/jsGL.mm [hunk 1]
```diff
@@ -38,6 +38,39 @@
...
+	context_2d *ctx = GET_CONTEXT_2D();
+	context_2d_resize(ctx, w, h);
+	texture_2d *tex = texture_manager_get_texture(texture_manager_get(), ctx->url);
+
+	JS::RootedValue glname(cx, JS::NumberValue(tex->name));
+	JS::RootedValue texurl(cx, CSTR_TO_JSVAL(cx, tex->url));
+
+	JSObject *tex_data = JS_NewObject(cx, nullptr, nullptr, nullptr);
+	JS_SetProperty(cx, tex_data, "__gl_name", glname);
+	JS_SetProperty(cx, tex_data, "_src", texurl);
```
Comment (jwilm) [inline tealeaf/js/jsGL.mm]: Why do we need to expose this to JS?

**Comment by sjrct** [comment]
Both are already exposed on Android. And symmetry aside, `resize` is already [made use of](https://github.com/sjrct/timestep/blob/5f95b69c3ac18e091d02fe8c50cfc3d413741bae/src/platforms/native/Canvas.js) in timestep, and `setTransform` is a [standard canvas 2d context function](http://www.w3.org/TR/2dcontext/#dom-context-2d-settransform).

**Comment by jwilm** [comment]
I think you answered the question of why `resize` and `setTransform` were exposed. I meant to ask about `__gl_name` and `_src` (note the position of the line note :grin:). Why did those need to be added to the otherwise empty return value object?

**Comment by sjrct** [comment]
My bad, if you look in [Context2D.js](https://github.com/gameclosure/timestep/blob/fd36cbd617fe906434a503bf8e413b6d6e9344f3/src/platforms/native/Context2D.js#L117-120) rather than Canvas.js as I mistakenly linked prior, timestep makes use of these values in order to determine what image is actually attached to each context. Native-android does the same.

File: tealeaf/js/jsGL.mm [hunk 2]
```diff
@@ -38,6 +38,39 @@
...
+	JSAG_ARG_INT32(h)
+
+	context_2d *ctx = GET_CONTEXT_2D();
+	context_2d_resize(ctx, w, h);
+	texture_2d *tex = texture_manager_get_texture(texture_manager_get(), ctx->url);
+
+	JS::RootedValue glname(cx, JS::NumberValue(tex->name));
+	JS::RootedValue texurl(cx, CSTR_TO_JSVAL(cx, tex->url));
+
+	JSObject *tex_data = JS_NewObject(cx, nullptr, nullptr, nullptr);
```
Comment (jwilm) [inline tealeaf/js/jsGL.mm]: This should be `JS::RootedObject tex_data(cx, JS_NewObject(cx, nullptr, nullptr, nullptr));` so the garbage collector knows about it.

**Comment by jwilm** [comment]
Makes sense. Thanks

**Pull request merged**